app/src/main/java/at/roteskreuz/stopcorona/model/managers/DatabaseCleanupManager.kt

```diff
@@ -72,23 +70,55 @@ class DatabaseCleanupManagerImpl(
         launch {
             val configuration = configurationRepository.observeConfiguration().blockingFirst()
 
-            val thresholdRedMessages = ZonedDateTime.now().minusHours(configuration.redWarningQuarantine?.toLong() ?: Long.MAX_VALUE)
-            infectionMessageDao.removeReceivedInfectionMessagesOlderThan(MessageType.InfectionLevel.Red, thresholdRedMessages)
-
-            val thresholdYellowMessages = ZonedDateTime.now().minusHours(configuration.yellowWarningQuarantine?.toLong() ?: Long.MAX_VALUE)
-            infectionMessageDao.removeReceivedInfectionMessagesOlderThan(MessageType.InfectionLevel.Yellow, thresholdYellowMessages)
-
-            val thresholdGreenMessages = ZonedDateTime.now().minusDays(THRESHOLD_REMOVE_INCOMING_GREEN_MESSAGES)
-            infectionMessageDao.removeReceivedInfectionMessagesOlderThan(MessageType.Revoke.Suspicion, thresholdGreenMessages)
+            val thresholdRedMessages = ZonedDateTime.now()
+                .minusHours(configuration.redWarningQuarantine?.toLong() ?: Long.MAX_VALUE)
```

Comment:
I would say in this case Long.MAX_VALUE is properly placed here because it's not a general default that might be applicable for all the cases where the redWarningQuarantine is used. For this case fits because we want to set the threshold in the past as much as possible.

Reply:
You will get `Invalid value for EpochDay (valid values -365243219162 - 365241780471): -384307168202263900` in case of `Long.MAX_VALUE`. If you want threshold as big as possible, set the time to epoch = 0.

Please search for other similar usages as well.

Reply:
Looking on the implementation of `minusHours()`, the operation seems possible. 
```
public ZonedDateTime minusHours(long hours) {
        return hours == -9223372036854775808L ? this.plusHours(9223372036854775807L).plusHours(1L) : this.plusHours(-hours);
    }
```

Reply:
Updated to avoid this invalid state